### ISSUE TITLE
Automatic Backup Fails

### ISSUE THREAD

**Issue by abrousseau001** [chegele/ValheimManager#15]
Good afternoon,

First off, I would like to say I appreciate your work on a manager application for Valheim. I am working on getting it setup on my server and I am having an issue with the automatic backup. My Windows Server 2022 is not domain joined so the user I am running the server on an account that has a space in the user's name (e.g. Bob Evans). I am assuming that is the issue, but I am not quite sure how the Node.js environment is setup to debug, so I can only offer what I believe to be the case. Please see the error message below:

```
[7/29/2023, 3:03:32 PM] ERROR   ValheimManager -  Unexpected error during automatic backup
ErrorError: Backup Failed: Error: ENOENT: no such file or directory, stat 'C:\Users\Bob Evan\AppData\LocalLow\IronGate\Valheim\worlds\SampleWorld.db'
```

If you have any advice on how to setup an environment to debug this issue I would be more than happy to if you cannot reproduce the issue on your end.

Thanks,

Adam Brousseau

**Comment by chegele** [comment]
Can you validate that the path exists by navigating to it directly in Windows File Explorer? This module is starting to get old, so the path where the game files are assumed to be could have changed. 

`C:\Users\Bob Evan\AppData\LocalLow\IronGate\Valheim\worlds\SampleWorld.db`

**Comment by abrousseau001** [comment]
That makes sense. I believe the latest dedicated server is using
world_local to avoid the previous backup to the cloud process. Assuming
that the directory just needs to be updated, how can I build the valheim
manager node server?  I know npm install installs the node dependencies
from the JSON, but how is it built? Is there a npm run task for that as
well?

**Comment by chegele** [comment]
No need to build the application. Node.js can run JavaScript files directly, which this project is written in. As you make changes to the installed files you can immediately test them by running the application again.

Here is the line you are looking for. Let me know if modifying it resolves your issues. 
[`this.worldsPath = path.join(os.homedir(), 'AppData/LocalLow/IronGate/Valheim/worlds');`](https://github.com/chegele/ValheimManager/blob/e10628c71b7adca05052e9d3bd778643412ec673/src/backups.js#L13C8-L13C8)

**Comment by abrousseau001** [comment]
Good evening. That was the line that needed to be changed. I just came back to Valheim after a year hiatus, and they are storing the files in the worlds_local directory instead. Would you like a PR for this update?

As for Node.js allowing the direct change of JavaScript, I just wasn't sure the configuration as different projects use compiled dependencies in some cases and others allow for the bundling of the application into a single binary. I use Webpack locally for a project that has scripts that execute Webpack to bundle my JavaScript files for an application I produce. With all the variants, I wasn't sure but appreciate you response.

**Comment by chegele** [comment]
Yes. Feel free to submit a pull request.

I will likely add some conditional logic to ensure that the manager is still compatible with earlier versions of Valheim in case anyone is choosing to not update.

Thanks for reaching out!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by abrousseau001** [comment]
Closed by #16
